src/foam/nanos/auth/CachingAuthService.java

```diff
@@ -82,59 +98,58 @@ public boolean checkUser(foam.core.X x, foam.nanos.auth.User user, String permis
     return permissionCheck;
   }
 
-  public void purgeCache(X x) {
-    User user = getUserFromContext(x);
-    if ( user != null ) {
-      userPermissionCache_.remove(user.getId());
+  public void purgeCache(Object obj) {
+    Long userId = null;
```

Comment:
Why Long instead of long?

Reply:
if the given obj is of a type we can pull a user id from, the Long will always be set as long cannot be null.  I used Long as it allows me to check null for not set.  I would have to define a value for user id that could be used as "not set" or use an additional boolean to determine this otherwise ... is there a better alternative?  Cause for concern?

Reply:
Commented use of long in https://github.com/kgrgreer/foam3/pull/857